internal/services/cdn/cdn_frontdoor_security_policy_resource.go

```diff
@@ -200,70 +208,78 @@ func resourceCdnFrontdoorSecurityPolicyRead(d *pluginsdk.ResourceData, meta inte
 	ctx, cancel := timeouts.ForRead(meta.(*clients.Client).StopContext, d)
 	defer cancel()
 
-	id, err := parse.FrontDoorSecurityPolicyID(d.Id())
+	id, err := securitypolicies.ParseSecurityPolicyID(d.Id())
 	if err != nil {
 		return err
 	}
 
-	resp, err := client.Get(ctx, id.ResourceGroup, id.ProfileName, id.SecurityPolicyName)
+	resp, err := client.Get(ctx, pointer.From(id))
 	if err != nil {
-		if utils.ResponseWasNotFound(resp.Response) {
+		if response.WasNotFound(resp.HttpResponse) {
 			d.SetId("")
 			return nil
 		}
 		return fmt.Errorf("retrieving %s: %+v", *id, err)
 	}
 
 	d.Set("name", id.SecurityPolicyName)
-	d.Set("cdn_frontdoor_profile_id", parse.NewFrontDoorProfileID(id.SubscriptionId, id.ResourceGroup, id.ProfileName).ID())
+	d.Set("cdn_frontdoor_profile_id", profiles.NewProfileID(id.SubscriptionId, id.ResourceGroupName, id.ProfileName).ID())
 
-	if props := resp.SecurityPolicyProperties; props != nil {
-		waf, ok := props.Parameters.AsSecurityPolicyWebApplicationFirewallParameters()
-		if !ok {
-			return fmt.Errorf("flattening %s: %s", id, "expected security policy web application firewall parameters")
-		}
+	model := resp.Model
 
-		// we know it's a firewall policy at this point,
-		// create the objects to hold the policy data
-		associations := make([]interface{}, 0)
+	if model == nil {
+		return fmt.Errorf("model is 'nil'")
+	}
```

Comment:
Here too, for the `if model := resp.Model; model != nil { d.Set() }` pattern

Reply:
Fixed.